db/migrations/20230129101723_fix_article_state.js

```diff
@@ -0,0 +1,15 @@
+exports.up = async (knex) => {
+  await knex.raw(/*sql*/ `
+  UPDATE article SET state = 'error' WHERE id IN (
+    SELECT article.id FROM draft 
+    LEFT JOIN article ON article.draft_id = draft.id
+      WHERE draft.publish_state != 'published' AND  article.draft_id IS NOT NULL
```

Comment:
all drafts that `publish_state` in `unpublished`, `error`, and `pending` are included?

Reply:
Yes, this is what the sql does.  You mean `draft.publish_state != 'published'` should be `draft.publish_state = 'error'` ?